### ISSUE TITLE
odd disagreement with anomaly and predicted

### ISSUE THREAD

**Issue by michaelneale** [cloudwalkio/omg-monitor#38]
![screen shot 2015-02-18 at 3 31 51 pm](https://cloud.githubusercontent.com/assets/14976/6242715/91d9b004-b788-11e4-8ec9-9a1148f32994.png)

For some reason the predicted value was spot on - but anomaly score was high - have you seen this?

**Comment by allanino** [comment]
This is weird. Never saw this before. What is your encoder resolution?

**Comment by michaelneale** [comment]
as in " 'seconds_per_request': 60 " - so 60 seconds - which may explain it. I was putting in data far more frequently than that (I assume that is what you meant?)

**Comment by allanino** [comment]
I don't think it is related to the rate of data feeding the model (but it would be nice to compute the processing time taken by each data point).

The resolution is also fine, but I do think it's related to the way the Random Distributed Scalar Encoder works. I think we should ask the NuPIC community for possible explanations.

**Comment by allanino** [comment]
I think the problem is odder yet:
![odd-monitor](https://cloud.githubusercontent.com/assets/5325241/6267700/6b21e12a-b82d-11e4-915d-fc69c2818510.png)
 
Though the anomalies scores settle down after a while, it keeps making predictions far off the actual values. This may be related to boosting, so I'll test this example without boosting and report back.

**Comment by allanino** [comment]
Without boosting we don't see those odd predictions, but the predictions never catch up the actual values, being always below it.

![odd-monitor-no-boost-same-window](https://cloud.githubusercontent.com/assets/5325241/6270985/d82ac784-b844-11e4-9d00-babc13dcffba.png)

**Comment by michaelneale** [comment]
yes I saw the same thing. I think you nailed something with the Scalar Encoder - I was also testing increating values - occasionally resetting (like a sawtooth wave) - some of it could be artifacts of it recalibrating as I break some limit for the value... which kind of forces a lot of change. 

I was surprised by the spikes in the anomaly score though - even in your one above - you can see it spiking even though it is hugging the real line nicely - *that* I still don't understand. The likelihood seems a bit less jumpy. 

How did you turn off the boosting BTW? (probably not something would normally want to do as the shape of the data is unknown - these are contrived examples).

**Comment by allanino** [comment]
I agree with you in that the high anomalies in perfect fitting is the oddest. I'll ask for help in NuPIC's mail list, as I cannot explain it with my limited knowledge.

For the boosting,  I setted it `maxBoost: 1.0` here: https://github.com/cloudwalkio/omg-monitor/blob/master/monitor/base_model_params.py#L56. Then rebuilt the image.

**Comment by michaelneale** [comment]
I think we can close this for now.

**Comment by michaelneale** [comment]
I think I am seeing this again:
![screen shot 2015-05-04 at 4 27 05 pm](https://cloud.githubusercontent.com/assets/14976/7449479/b00af7ea-f27a-11e4-898b-37cf335417b6.png)

This looks very similar to the boosting you were seeing. Do you run with boosting set to 2.0? Should this be a parameter. I will see if it settles down (you can see in my screen cap that it does for a bit, btu then tries again). You had problems with this elsewhere? This is ostensibly cyclical data (goes up and down int the day) and (I think) relatively noise free.

**Comment by michaelneale** [comment]
Its possible that maxBoost may not be suitable for this type of time series: http://markmail.org/message/cvastzj3ugiykzy2#query:+page:1+mid:u6cdqnpnjr3gnxp6+state:results (I need to try some experiments). I gather being a float it doesn't have to be integral (although I thought it was number of columns, but maybe is a ratio? hard to find out what it means exactly).

**Comment by allanino** [comment]
Hi, @michaelneale! 

That's a nice explanation from the mail list. I think we can create an option to override default model parameters, helping us on debugging this issue and giving more flexibility to the service. 

I'll create another issue for that, but if you don't want to wait for that implementation (I should get it done by the end of the week), you can just set `maxBoost: 1.0` in [monitor/base_model_params.py#L56](https://github.com/cloudwalkio/omg-monitor/blob/master/monitor/base_model_params.py#L56) and rebuild the image. If you do that, let me know your results.

**Comment by michaelneale** [comment]
I will try that today. I have let it stay overnight to see if it settles
downy his certainly matches up with what subutai said - it is a fairly
"plain" time series - not a lot of patterns so it is likely columns are
being under utilised. I may try a few values.

**Comment by michaelneale** [comment]
So far this seems much better, on the whole. I am going to try a maxBoost=1.1 experiment too - for comparison. There are some things it seems to be struggling with - taking a little longer to adapt.

**Comment by michaelneale** [comment]
maxBoost=1.1 seems to work quite nicely for me needs as far as I can see, I think I will settle with that for now.

**Comment by allanino** [comment]
That's nice! I'll create another issue for the parameters override task, to that we can set this on the configuration file. After that I'll play with maxBoost on my own containers to see it I can get improvements too.

**Comment by allanino** [comment]
Hi, @michaelneale! I just fixed #48, so now we can specify any options under `modelParams` in [monitor/base_model_params.py#L33](https://github.com/cloudwalkio/omg-monitor/blob/master/monitor/base_model_params.py#L33) in the config YAML file itself, like this:

```
nupic_model_params:
    spParams:
        maxBoost: 1.1
    tpParams:
        activationThreshold: 16
```

I haven't tested it with your `Dynamic` class, but the `Monitor` class is ready to take a `nupic_model_params` dictionary with keys to update in your config, like this:

```
{
OTHER_CONFIG_KEYS,
'nupic_model_params': {'spParams': {'maxBoost': 1.1},
                                       'tpParams': {'activationThreshold': 16}}
}
```

**Comment by michaelneale** [comment]
that's great. BTW did you try it with some other values? I have you changed the activationThreshold for the temporal pooler yourself?

**Comment by allanino** [comment]
I haven't tried using new boost values extensively. About the `activationThreshold`, it's just an example with the default value, which is `16`.
